starlette/requests.py

```diff
@@ -103,6 +103,8 @@ def cookies(self) -> typing.Dict[str, str]:
     async def json(self) -> typing.Any:
         if not hasattr(self, "_json"):
             body = await self.body()
+            if not body:
+                body = b"{}"
```

Comment:
You could probably do something like:
```suggestion
                if "app" in self._scope:
                    raise HTTPException(status_code=400)
                return PlainTextResponse("Bad Request", status_code=400)
```